packages/flutter/lib/src/widgets/image.dart

```diff
@@ -0,0 +1,251 @@
+
+  /// How to paint any portions of the layout bounds not covered by the image.
+  final ImageRepeat repeat;
+
+  /// The center slice for a nine-patch image.
+  ///
+  /// The region of the image inside the center slice will be stretched both
+  /// horizontally and vertically to fit the image into its destination. The
+  /// region of the image above and below the center slice will be stretched
+  /// only horizontally and the region of the image to the left and right of
+  /// the center slice will be stretched only vertically.
+  final Rect centerSlice;
+
+  /// Whether to continue showing the old image (true), or briefly show nothing
+  /// (false), when the image provider changes.
+  // TODO(ianh): Find a better name.
+  final bool gaplessPlayback;
+
+  @override
+  _ImageState createState() => new _ImageState();
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    description.add('image: $image');
+    if (width != null)
+      description.add('width: $width');
+    if (height != null)
+      description.add('height: $height');
+    if (color != null)
+      description.add('color: $color');
+    if (fit != null)
+      description.add('fit: $fit');
+    if (alignment != null)
+      description.add('alignment: $alignment');
+    if (repeat != ImageRepeat.noRepeat)
+      description.add('repeat: $repeat');
+    if (centerSlice != null)
+      description.add('centerSlice: $centerSlice');
+  }
+}
+
+class _ImageState extends State<Image> {
+  ImageStream _imageStream;
+  ImageInfo _imageData;
+
+  @override
+  void initState() {
+    super.initState();
+    _resolveImage();
+  }
+
+  @override
+  void didUpdateConfig(Image oldConfig) {
+    if (config.image != oldConfig.image)
+      _resolveImage();
+  }
+
+  @override
+  void dependenciesChanged() {
+    _resolveImage();
+    super.dependenciesChanged();
+  }
+
+  @override
+  void dispose() {
+    _imageStream.removeListener(_handleImageChanged);
+    super.dispose();
+  }
+
+  void _resolveImage() {
+    final ImageStream oldImageStream = _imageStream;
+    _imageStream = config.image.resolve(createLocalImageConfiguration(
+      context,
+      size: config.width != null && config.height != null ? new Size(config.width, config.height) : null
+    ));
+    assert(_imageStream != null);
+    if (_imageStream.key != oldImageStream?.key) {
+      oldImageStream?.removeListener(_handleImageChanged);
+      if (!config.gaplessPlayback)
+        setState(() { _imageData = null; });
+      _imageStream.addListener(_handleImageChanged);
+    }
+  }
+
+  void _handleImageChanged(ImageInfo imageData) {
+    setState(() {
+      _imageData = imageData;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new RawImage(
+      image: _imageData?.image,
+      width: config.width,
+      height: config.height,
+      scale: _imageData == null ? 1.0 : _imageData.scale,
```

Comment:
i've changed this because it looks nicer, but that is technically one extra branch!